### PR TITLE
tyr: fix status 404 in instance model method

### PR DESCRIPTION
**Pull request by francois-vincent**
tyr tests now OK

### CODE REVIEW

File: source/navitiacommon/navitiacommon/models.py [hunk 1]
```diff
@@ -346,11 +346,14 @@ def get_by_name(cls, name):
     @classmethod
     def get_from_id_or_name(cls, id=None, name=None):
         if id:
-            return cls.query.get_or_404(id)
```
Comment (antoine-de) [inline source/navitiacommon/navitiacommon/models.py:349]: I don't understand the difference ?

Comment (pbougue) [inline source/navitiacommon/navitiacommon/models.py:349]: It's weird that it's not dealing that thing, yes. The "None" case is ok for flask?
Anyway the code is ok for me, just it's weird...

Comment (francois-vincent) [inline source/navitiacommon/navitiacommon/models.py:349]: What is weird is that flask-sqlalchemy abort() does not complies with falsk-restful resource (flask_restful.Resource).
Flask builds on external modules, possibly not cross-consistent as they are external and independant.
Django is better for consistency indeed !

**Comment by francois-vincent** [comment]
Jenkins, test this please

**Comment by francois-vincent** [comment]
Jenkins, test this please

**Comment by francois-vincent** [comment]
Jenkins, test this please

**Comment by francois-vincent** [comment]
Jenkins, test this please

**Comment by francois-vincent** [comment]
Jenkins, test this please

**Comment by francois-vincent** [comment]
Jenkins, test this please

**Comment by francois-vincent** [comment]
Jenkins, test this please

**Comment by francois-vincent** [comment]
Jenkins, test this please

**Comment by pbougue** [comment]
Jenkins, test this please

**Comment by pbougue** [comment]
TODO: remove the try in delete and put in resources.py -> done in another PR

**Pull request closed without merging**